api/src/services/report/validate.js

```diff
@@ -1,67 +1,37 @@
-const _ = require('underscore');
```

Comment:
this whole thing was way more complicated than it needed to be because of undocumented support for dot notation in field declarations (`foo.bar` as a field name would be mapped into `foo: {bar: ..}}` in JSON.